beacon-chain/p2p/discovery.go

```diff
@@ -99,18 +99,22 @@ func startDHTDiscovery(host core.Host, bootstrapAddr string) error {
 }
 
 func parseBootStrapAddrs(addrs []string) (discv5Nodes []string, kadDHTNodes []string) {
-	for _, addr := range addrs {
-		_, err := enode.Parse(enode.ValidSchemes, addr)
-		if err == nil {
-			discv5Nodes = append(discv5Nodes, addr)
-			continue
-		}
-		_, err = multiAddrFromString(addr)
-		if err == nil {
-			kadDHTNodes = append(kadDHTNodes, addr)
-			continue
+	if len(addrs) == 1 && addrs[0] == "" {
+		log.Warn("No bootstrap addresses supplied")
+	} else {
+		for _, addr := range addrs {
+			_, err := enode.Parse(enode.ValidSchemes, addr)
+			if err == nil {
+				discv5Nodes = append(discv5Nodes, addr)
+				continue
+			}
+			_, err = multiAddrFromString(addr)
+			if err == nil {
+				kadDHTNodes = append(kadDHTNodes, addr)
+				continue
+			}
+			log.Errorf("Invalid bootstrap address of %s provided", addr)
 		}
-		log.Errorf("Invalid bootstrap address of %s provided", addr)
 	}
```

Comment:
can you maybe instead check it at the end ? 
```go
if len(discv5Nodes) == 0 && len(kadDHTNodes) == 0 {
     log.Warn("No valid bootstrap addresses supplied")
}
```

Reply:
If we do that we'd get an error during attempting to parse an empty entry "" as well as the warning.  So to avoid that we'd have an additional case inside the loop; you okay with that?

Reply:
ah yeap, that would log the error for `""` . Sounds good to me to handle that case

Reply:
Update the code; it does a better job of logging the errors and warnings at the right time.  Thanks.